crates/vm/levm/src/memory.rs

```diff
@@ -64,16 +65,25 @@ impl Memory {
         self.data[dest_offset..dest_offset + size].copy_from_slice(&temp);
     }
 
-    pub fn expansion_cost(&self, memory_byte_size: usize) -> usize {
+    pub fn expansion_cost(&self, memory_byte_size: usize) -> Result<usize, VMError> {
         if memory_byte_size <= self.data.len() {
-            return 0;
+            return Ok(0);
         }
-        let new_memory_size_word = (memory_byte_size + WORD_SIZE - 1) / WORD_SIZE;
-        let new_memory_cost =
-            (new_memory_size_word * new_memory_size_word) / 512 + (3 * new_memory_size_word);
+
+        let new_memory_size_word = memory_byte_size
+            .checked_add(WORD_SIZE - 1)
+            .ok_or(VMError::OverflowInArithmeticOp)?
+            / WORD_SIZE;
+
+        let new_memory_cost = new_memory_size_word
+            .checked_mul(new_memory_size_word)
+            .and_then(|square| square.checked_div(MEMORY_EXPANSION_QUOTIENT))
```

Comment:
This `checked_div` is not necessary as `MEMORY_EXPANSION_QUOTIENT` is not 0.

Reply:
Done!